plugin-server/src/main/pluginsServer.ts

```diff
@@ -152,6 +152,28 @@ export async function startPluginsServer(
         Sentry.captureException(error)
     })
 
+    process.on('uncaughtException', async (error: Error) => {
+        // If there are unhandled exceptions anywhere, perform a graceful
+        // shutdown. The initial trigger for including this handler is due to
+        // the graphile-worker code throwing an exception when it can't call
+        // `nudge` on a worker. Unsure as to why this happens, but at any rate,
+        // to ensure that we gracefully shutdown Kafka consumers, for which
+        // unclean shutdowns can cause considerable delay in starting to consume
+        // again, we try to gracefully shutdown.
+        //
+        // See https://nodejs.org/api/process.html#event-uncaughtexception for
+        // details on the handler.
+        status.error('🤮', `uncaught_exception`, { error: error.stack })
+
+        if (error.message === 'nudge called after worker terminated') {
+            // Be specific about the error, otherwise we appear to end up in a
+            // recursive situation.
+            await closeJobs()
+        }
+
+        process.exit(1)
```

Comment:
what would be the implications of us being more graceful on any uncaught exception you think? 🤔 

i.e. make sure everything flushes on any uncaughts

Reply:
Yeah that was my first attempt, then I had more error logging then I decided to be more specific. I think there needs to be another iteration but I'd love to have some input from node experts that could test this out locally and maybe tell me what to do 😆

Reply:
I would love to put some tests on this because I feel like I don't know what's going on btw